S3/FileLists.py

```diff
@@ -60,34 +60,37 @@ def handle_exclude_include_walk_dir(root, dirname):
     Should this root/dirname directory be excluded? (otherwise included by default)
     Exclude dir matches in the current directory
     This prevents us from recursing down trees we know we want to ignore
-    return True for including, and False for excluding
+    return True for excluding, and False for including
     '''
     cfg = Config()
+    # python versions end their patterns (from globs) differently, test for different styles
+    directory_patterns = [u'\\/$', u'\\/\\Z(?ms)', u'/)$', u'/)\\Z']
+
     d = os.path.join(root, dirname, '')
     debug(u"CHECK: %r" % d)
     excluded = False
     for r in cfg.exclude:
-        # python versions end their patterns (from globs) differently, test for both styles.
-        if not (r.pattern.endswith(u'\\/$') or r.pattern.endswith(u'\\/\\Z(?ms)')): continue # we only check for directory patterns here
+        if not any(r.pattern.endswith(dp) for dp in directory_patterns): continue # we only check for directory patterns here
+        debug(u"EXCL-TEST: %s ~ %s" % (d, r.pattern))
```

Comment:
Maybe this new debug is not necessary once it works .

Reply:
1) Here I just copied same logic for debug messages as in current code, see INCL-TEST/INCL-MATCH messages below on lines 81 and 84 (old numeration). 
2) in general, all debugs are unnecessary when everything works fine :) This one was really helpful when I was trying to figure out what happens: does program escape from cycle on `if not any(r.pattern.endswith... continue` one line above, or we do check regexps, but they just don't match.

Reply:
I don't like very much the log here because, in the other cases, we have already an "exclusion / inclusion Match" before printing a log.


The EXCL-TEST would print the line for all test for all directories tested.
In my opinion it is only useful to debug the current issue, and could be temporary added like now to debug if necessary.

And, in the end, you will have the PASS and EXCLUDE log that will give you an hint if something does not go as expected.

Reply:
ok, removed